src/applet.c

```diff
@@ -335,6 +355,18 @@ static void brisk_menu_applet_settings_changed(GSettings *settings, const gchar
 }
 
 /**
+ * Update the applet label text in accordance with settings
+ */
+void brisk_menu_applet_update_label_text(BriskMenuApplet *self, gchar *key)
+{
+        if (g_str_equal(key, "")) {
+                gtk_label_set_text(GTK_LABEL(self->label), _("Menu"));
+        } else {
+                gtk_label_set_text(GTK_LABEL(self->label), key);
```

Comment:
This will **always** result in the custom key setting `label-key` as the display string. You're setting with the key and not the actual label. Test locally you'll see what I mean, when you update with dconf *after* brisk is started.